src/components/search-bar/search-bar.jsx

```diff
@@ -16,14 +18,34 @@ import './search-bar.css';
 export default class SearchBar extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      query: {
+        url: null,
+        startDate: null,
+        endDate: null
+      }
+    };
     this._didSearch = this._didSearch.bind(this);
     this._dispatchSearch = debounce(this._dispatchSearch.bind(this), 500);
+    this.urlSearch = this.urlSearch.bind(this);
+    this.dateSearch = this.dateSearch.bind(this);
+  }
+
+  componentDidUpdate(previousProps, previousState) {
+    if (previousState.query !== this.state.query) {
+      this.props.onSearch(this.state.query);
+    }
   }
 
   _didSearch (event) {
     this._dispatchSearch(event.target.value);
   }
 
+  /**
+   * Parses url to generate wildcard pattern for url search query and then calls urlSearch
+   *
+   * @param {String} url
+   */
   _dispatchSearch (url) {
```

Comment:
The `dispatch` naming here existed because this is the function that *dispatches calls to `onSearch`*. Since that’s no longer true, this should probably be renamed. I’m thinking this function is actually what `urlSearch` should be.

(`_didSeach` is also no longer a very clear name now, either! Maybe `didChangeUrl` or `handleUrlInput` or something?)

Also, if you’re going to add doc comments here (and below, on `urlSearch`), they should include `@private` since this is a private function.